pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -273,3 +304,61 @@ func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitR
 		Message: "Node is experiencing CPU Pressure",
 	}
 }
+
+const (
+	reason  = "Evicted"
+	message = "The node no longer has CPUs available to run the pod"
+)
+
+func isNoCPUPod(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if isNoCPUContainer(&container) {
+			return true
+		}
+	}
+	return false
+}
+
+func isNoCPUContainer(container *v1.Container) bool {
+	for resourceName, request := range container.Resources.Requests {
+		if resourceName == v1.ResourceCPU && !request.IsZero() {
+			return false
+		}
+	}
+	return true
+}
+
+func getNoCPUPods(pods []*v1.Pod) []*v1.Pod {
+	var noCPUPods []*v1.Pod
+	for _, pod := range pods {
+		if isNoCPUPod(pod) {
+			noCPUPods = append(noCPUPods, pod)
+		}
+	}
+	return noCPUPods
+}
+
+func (m *manager) evictNoCPUPods() error {
+	noCPUPods := getNoCPUPods(m.getPodsFunc())
+	if len(noCPUPods) == 0 {
+		return nil
+	}
+
+	glog.Infof("[cpumanager] attempting to evict pods with no CPU request because shared CPU pool is empty: %v", noCPUPods)
```

Comment:
Nit: technically we are evicting because there is CPU pressure (although with static policy this only happens when shared pool is empty)